sceptre/stack.py

```diff
@@ -196,9 +196,10 @@ def create(self):
             "Tags": [
                 {"Key": str(k), "Value": str(v)}
                 for k, v in self.config.get("stack_tags", {}).items()
-            ],
-            "OnFailure": self.config.get("on_failure", 'ROLLBACK')
+            ]
         }
+        if self.config.get("on_failure"):
+            self.config['OnFailure'] = self.config.get("on_failure") 
```

Comment:
Not sure why you've referenced `self.config['OnFailure'] =`?
Surely you want something like this:
`create_stack_kwargs.update({"OnFailure": self.config.get("on_failure")})`